library/std/src/os/unix/net/ancillary.rs

```diff
@@ -407,7 +434,13 @@ pub struct SocketAncillary<'a> {
 }
 
 impl<'a> SocketAncillary<'a> {
-    /// Create an ancillary data with the given buffer.
+    /// Create a struct for accessing an ancillary data packet in `buffer`.
```

Comment:
> accessing [...] in buffer

While this isn't wrong I think it could be misunderstood that the purpose of this constructor is to pass in a buffer that already contains messages and then deserializing it.

The next sentence clarifies it but I think we can still do better. Maybe something like

> Create a struct holding ancillary data. `buffer` is used to store the serialized data without dynamic allocations.

Exposing the buffer is mostly an optimization to play nice with low-level network code that uses preallocated buffers.

Reply:
I'm trying to avoid saying that the struct holds ancillary data because it doesn't.

Reply:
>  Create a struct to process ancillary data. `buffer` is used to store the serialized data without dynamic allocations.